### ISSUE TITLE
[BUG] Keep the same commit SHA between workflows

### ISSUE THREAD

**Issue by rlinstorres** [tj-actions/changed-files#543]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this issue exist in the latest version?

- [X] I'm using the latest release

### Describe the bug?

Hi, first of all, I am not sure if it is a bug or an improvement.

I have a workflow triggered by `push` events and the result of this workflow is triggering another workflow by `workflow_run`. The problem is in my second workflow when triggered the `base_sha` value is not keeping the same value as the first workflow.

Note: It's happening when I have more than one PR merged into the master branch sequentially.

First workflow:
```yaml
jobs:
  job:
    name: Job
    runs-on: self-hosted-runner
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Modified Files
        uses: tj-actions/changed-files@v23.1
        with:
          files: .github/workflows/test.txt
```

Second workflow:
```yaml
jobs:
  condition:
    if: github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch'
    name: Job Condition
    runs-on: self-hosted-runner
    steps:
      - uses: actions/checkout@v3
        name: github_event_workflow_run_head_sha
        with:
          fetch-depth: 0
          ref: ${{ github.event.workflow_run.head_sha }}

      - name: Modified Files
        uses: tj-actions/changed-files@v23.1
        with:
          files: .github/workflows/test.txt
          sha: ${{ github.event.workflow_run.head_sha }}
```

### To Reproduce

For example:

First PR merged and first workflow:
Commit SHA: `4304a00cd43101d81d49b9ce277459267c446a5d`
Base commit SHA: `8668dbc37408aafb315621ee66bdb154bfafc1ce`

<img width="1003" alt="Screen Shot 2022-07-05 at 5 11 10 PM" src="https://user-images.githubusercontent.com/11336131/177359966-2bddab11-2429-4e93-ab1a-e2b53f0e2641.png">

Second workflow triggered from the first workflow:

Commit SHA: `4304a00cd43101d81d49b9ce277459267c446a5d`
Base commit SHA: `4304a00cd43101d81d49b9ce277459267c446a5d`

<img width="1017" alt="Screen Shot 2022-07-05 at 5 13 20 PM" src="https://user-images.githubusercontent.com/11336131/177360395-036a32c2-04a7-482c-909f-e6a7f3826c16.png">

### What OS are you seeing the problem on?

ubuntu-latest or ubuntu-20.04

### Expected behavior?

I would like to keep the same `base_sha` value when working with `workflow_run` as a trigger.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jackton1** [comment]
@rlinstorres Can you also include the event that triggers the workflow in your description.

Example
```yml
on:
   ...

jobs:
  job:
    name: Job
    runs-on: self-hosted-runner
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Modified Files
        uses: tj-actions/changed-files@v23.1
        with:
          files: .github/workflows/test.txt
```

**Comment by rlinstorres** [comment]
> 

For sure, sorry!

- First workflow:
```yaml
on:
  push:
    branches:
      - master
  workflow_dispatch:
```

- Second workflow:
```yaml
on:
  workflow_run:
    workflows: [first-workflow]
    types: [completed]
    branches:
      - master
  workflow_dispatch:
```

**Comment by rlinstorres** [comment]
> @rlinstorres Can you also include the event that triggers the workflow in your description.
> 
> Example
> 
> ```yaml
> on:
>    ...
> 
> jobs:
>   job:
>     name: Job
>     runs-on: self-hosted-runner
>     steps:
>       - uses: actions/checkout@v3
>         with:
>           fetch-depth: 0
> 
>       - name: Modified Files
>         uses: tj-actions/changed-files@v23.1
>         with:
>           files: .github/workflows/test.txt
> ```

For sure, sorry!

- First workflow:
```yaml
on:
  push:
    branches:
      - master
  workflow_dispatch:
```

- Second workflow:
```yaml
on:
  workflow_run:
    workflows: [first-workflow]
    types: [completed]
    branches:
      - master
  workflow_dispatch:
```

**Comment by jackton1** [comment]
@rlinstorres This should be resolved in the latest release [v23.2](https://github.com/tj-actions/changed-files/releases/tag/v23.2)
